go-controller/pkg/util/net_unit_test.go

```diff
@@ -278,3 +279,115 @@ func TestMatchIPFamily(t *testing.T) {
 		})
 	}
 }
+
+func TestJoinHostPortInt32(t *testing.T) {
+	tests := []struct {
+		desc    string
+		inpHost string
+		inpPort int32
+		outExp  string
+	}{
+		{
+			desc:    "empty string for host name and a negative port number",
+			inpPort: -25,
+			outExp:  ":-25",
+		},
+		{
+			desc:    "valid non-zero length string for host name",
+			inpHost: "hostname",
+			inpPort: 22,
+			outExp:  "hostname:22",
+		},
+		{
+			desc:    "valid IPv4 address as host name",
+			inpHost: "192.168.1.15",
+			inpPort: 22,
+			outExp:  "192.168.1.15:22",
+		},
+		{
+			desc:    "valid IPv6 address as host name",
+			inpHost: "fd01::1234",
+			inpPort: 22,
+			outExp:  "[fd01::1234]:22",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			res := JoinHostPortInt32(tc.inpHost, tc.inpPort)
```

Comment:
Now this actual in the code is just a wrapper around the net library function net.JoinHostPort and your unit testing, in this case, are testing the library function... So value this adds here?

Reply:
@aneeshkp you are right in your observation, at this time, I am adding coverage for all (where appropriate) functions defined in net.go so that any refactor later would ensure tests pass or get appropriately reworked to match the refactor.